ddclient

```diff
@@ -2345,6 +2321,7 @@ sub extract_ipv6 {
                 next
             }
         }
+        $parsed =~ s/\b0+\B//g; ## remove embedded leading zeros
```

Comment:
This worries me a bit because I believe leading zeros are fairly common in real-world IPv6 addresses. I don't want `is_ipv6` to reject a legitimate IPv6 address unless the format is highly unusual. Maybe leave this out for now?

Reply:
Well, I think that should be resolved in `is_ipv6` and not in `extract_ipv6`.  The purpose of this function is to obtain the IPv6 address from a larger body of text (which could in fact be nothing more than an IP address) for use elsewhere in the program... e.g. to send up to a dynamic DNS hosting site.  I think that `extract_ipv6` should return the _best_ format for that purpose and I think embedded leading zeros should be removed.

Now I do agree that this is problematic for `is_ipv6` which now fails on a perfectly fine IPv6 address.  This really only matters for an address manually set by the user in `ip=<address>` or `ipv6=<address>`, if you remember you had me switch from using `extract_ipv6` over there to using `is_ipv6`.   So I would rather fix this by either running manually entries through extract first, or  changing `is_ipv6` to accept addresses with embedded leading zeros.  I think I prefer updating `is_ipv6` to accept them -- let me propose a change to that and see what you think.

Whatever we do we should do the same for IPv4.

Reply:
> I think I prefer updating is_ipv6 to accept them

Yup, I agree.

> Whatever we do we should do the same for IPv4.

I disagree. I'm OK with rejecting leading zeros for IPv4 because they're extremely rare nowadays and can be ambiguous (some systems interpret leading zeros in an IPv4 address as signifying octal). For IPv6, leading zeros does not cause ambiguity; it's just not as compact.

Reply:
Take a look at the change I just committed to accept embedded leading zeros within the IPv6 address.

Reply:
> Take a look at the change I just committed to accept embedded leading zeros within the IPv6 address.  This is another place that the`/r` regex modifier would have been useful:-)

Reply:
> I disagree. I'm OK with rejecting leading zeros for IPv4 because they're extremely rare nowadays and can be ambiguous (some systems interpret leading zeros in an IPv4 address as signifying octal). For IPv6, leading zeros does not cause ambiguity; it's just not as compact.

Fine with me.